third_party/terraform/resources/resource_cloudiot_registry.go

```diff
@@ -369,3 +471,79 @@ func resourceCloudIoTRegistryStateImporter(d *schema.ResourceData, meta interfac
 	d.SetId(id)
 	return []*schema.ResourceData{d}, nil
 }
+
+func resourceCloudIotRegistryStateUpgradeV0toV1(rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+	log.Printf("[DEBUG] State before upgrade: %#v", rawState)
+
+	oldCfg, ok := rawState["event_notification_config"]
+	if ok {
+		delete(rawState, "event_notification_config")
+	}
+
+	v, ok := rawState["event_notification_configs"]
+	if ok && v != nil {
+		// Just ignore old value if event_notification_configs is already set.
```

Comment:
Can the value already be set? As I understand, the upgrade will run during the first refresh upon using a new provider version?